File: samples/jvm/src/main/java/dev/pubgkt/MasterySample.java
package dev.pubgkt;

import java.util.List;

import kotlin.coroutines.EmptyCoroutineContext;
import kotlinx.coroutines.BuildersKt;

import dev.pubgkt.mastery.GetSurvivalMasteryByAccountIdKt;
import dev.pubgkt.mastery.GetWeaponMasteryByAccountIdKt;
import dev.pubgkt.mastery.SurvivalMastery;
import dev.pubgkt.mastery.WeaponMastery;
import dev.pubgkt.players.GetPlayersByNameKt;
import dev.pubgkt.players.Player;

public class MasterySample {
    public static void main(String[] args) throws InterruptedException {
        if (args.length == 0) {
            throw new IllegalArgumentException("Usage: MasterySample <api-key>");
        }

        PubgApi api = new PubgApi(args[0]);

        List<Player> players = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetPlayersByNameKt.getPlayersByNames(api, List.of("sparkingg"), Platform.STEAM, cont)
        );
        String accountId = players.get(0).getId();

        System.out.println("=== getWeaponMasteryByAccountId ===");
        WeaponMastery weaponMastery = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetWeaponMasteryByAccountIdKt.getWeaponMasteryByAccountId(api, accountId, Platform.STEAM, cont)
        );
        System.out.println("Player: " + weaponMastery.getPlayerId() + ", Platform: " + weaponMastery.getPlatform());

        System.out.println("\n=== getSurvivalMasteryByAccountId ===");
        SurvivalMastery survivalMastery = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetSurvivalMasteryByAccountIdKt.getSurvivalMasteryByAccountId(api, accountId, Platform.STEAM, cont)
        );
        System.out.println("Player: " + survivalMastery.getPlayerId());
        System.out.println("XP: " + survivalMastery.getXp() + ", Level: " + survivalMastery.getLevel() + ", Tier: " + survivalMastery.getTier());
        System.out.println("Total matches: " + survivalMastery.getTotalMatchesPlayed());
    }
}

File: samples/jvm/src/main/kotlin/dev/pubgkt/MasterySample.kt
package dev.pubgkt

import dev.pubgkt.mastery.getSurvivalMasteryByAccountId
import dev.pubgkt.mastery.getWeaponMasteryByAccountId
import dev.pubgkt.players.getPlayersByNames

suspend fun main(args: Array<String>) {
    val apiKey = args.getOrNull(0) ?: error("Usage: MasterySample <api-key>")
    val api = PubgApi(apiKey)

    val players = api.getPlayersByNames("sparkingg")
    val accountId = players.first().id

    println("=== getWeaponMasteryByAccountId ===")
    val weaponMastery = api.getWeaponMasteryByAccountId(accountId)
    println("Player: ${weaponMastery.playerId}, Platform: ${weaponMastery.platform}")

    println("\n=== getSurvivalMasteryByAccountId ===")
    val survivalMastery = api.getSurvivalMasteryByAccountId(accountId)
    println("Player: ${survivalMastery.playerId}")
    println("XP: ${survivalMastery.xp}, Level: ${survivalMastery.level}, Tier: ${survivalMastery.tier}")
    println("Total matches: ${survivalMastery.totalMatchesPlayed}")
    println("Top 10s: ${survivalMastery.top10}")
}
